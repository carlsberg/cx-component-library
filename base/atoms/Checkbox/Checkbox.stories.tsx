import { Story } from '@storybook/react';
import React, { useState } from 'react';
import { Checkbox } from './Checkbox';
import { CheckboxProps } from './Checkbox.types';

export default {
  title: 'Atoms/Checkbox',
  component: Checkbox,
  argTypes: {}
};

const Template: Story<CheckboxProps> = ({ value }: CheckboxProps) => {
  const [stateChecked, setStateChecked] = useState(true);
  return (
    <Checkbox
      value={value}
      labelText="Label text"
      checked={stateChecked}
      error="Error text"
      onValueChange={() => setStateChecked(!stateChecked)}
    />
  );
};

export const Main = Template.bind({});
Main.args = {
  value: 'Test value'
};
