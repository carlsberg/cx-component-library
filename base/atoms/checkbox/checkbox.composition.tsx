import React from 'react';
import { Checkbox } from './checkbox';

// sets the Component preview in gallery view
export const BasicText = () => <Checkbox onValueChange={() => null} value="Test" checked />;
