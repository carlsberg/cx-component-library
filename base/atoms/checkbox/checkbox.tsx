import React from 'react';
import {
  StyledCheckboxContainer,
  StyledCheckboxDisplayInput,
  StyledCheckboxHiddenInput,
  StyledCheckboxLabel,
  StyledCheckboxLabelText,
  StyledError
} from './checkbox.styled';
import { CheckboxProps } from './checkbox.types';

export const Checkbox = ({ id, value, checked, labelText, error, onValueChange }: CheckboxProps) => (
  <StyledCheckboxContainer>
    <StyledCheckboxLabel htmlFor={id}>
      <StyledCheckboxHiddenInput
        id={id}
        value={value}
        onChange={(e) => onValueChange(!(e.target as HTMLInputElement).checked)}
        checked={checked}
      />
      <StyledCheckboxDisplayInput checked={checked} />
      <StyledCheckboxLabelText>{labelText}</StyledCheckboxLabelText>
    </StyledCheckboxLabel>
    {error && <StyledError>{error}</StyledError>}
  </StyledCheckboxContainer>
);
