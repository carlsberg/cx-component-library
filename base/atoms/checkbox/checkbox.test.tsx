import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { Checkbox } from './checkbox';

describe('checkbox', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Checkbox labelText="Label text" value="test" onValueChange={() => null} checked />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders elements', () => {
    const mockFn = jest.fn();
    render(<Checkbox labelText="Label text" error="Error text" value="Test value" onValueChange={mockFn} checked />);
    expect(screen.getByLabelText('Label text')).toBeInTheDocument();
    expect(screen.getByText('Error text')).toBeInTheDocument();
    expect(screen.getByDisplayValue('Test value')).toBeInTheDocument();
  });

  it('calls function on click', () => {
    const mockFn = jest.fn();
    render(<Checkbox labelText="Label text" error="Error text" value="Test value" onValueChange={mockFn} checked />);
    const checkbox = screen.getByDisplayValue('Test value');
    fireEvent.click(checkbox);
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
