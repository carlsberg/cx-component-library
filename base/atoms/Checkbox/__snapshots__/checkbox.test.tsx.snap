// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox renders correctly 1`] = `
<div
  className="sc-bdnxRM UOsks"
>
  <label
    className="sc-hKFxyN kChuaK"
  >
    <input
      checked={true}
      className="sc-eCApnc YhDET"
      onChange={[Function]}
      type="checkbox"
      value="test"
    />
    <div
      checked={true}
      className="sc-jSFjdj eUvaQc"
    />
    <span
      className="sc-dlnjwi cnyngY"
    >
      Label text
    </span>
  </label>
</div>
`;
